crates/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -282,6 +282,122 @@ pub(super) fn temporal_range_dispatch(
         _ => {},
     };
 
+    let start = start.get(0).unwrap().extract::<i64>().unwrap();
+    let stop = stop.get(0).unwrap().extract::<i64>().unwrap();
+
+    let out = match dtype {
+        DataType::Date => date_range_impl(
+            name,
+            start,
+            stop,
+            every,
+            closed,
+            TimeUnit::Milliseconds,
+            None,
+        )?,
+        DataType::Datetime(tu, ref tz) => {
+            date_range_impl(name, start, stop, every, closed, tu, tz.as_ref())?
+        },
+        DataType::Time => date_range_impl(
+            name,
+            start,
+            stop,
+            every,
+            closed,
+            TimeUnit::Nanoseconds,
+            None,
+        )?,
+        _ => unimplemented!(),
+    };
+    Ok(out.cast(&dtype).unwrap().into_series())
+}
+
+pub(super) fn temporal_ranges_dispatch(
```

Comment:
This introduces quite some code duplication, which will be addressed in https://github.com/pola-rs/polars/pull/10213